GiniVision/Classes/Core/GiniScreenAPICoordinator.swift

```diff
@@ -222,9 +217,15 @@ extension GiniScreenAPICoordinator: UINavigationControllerDelegate {
                               from fromVC: UIViewController,
                               to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
         if fromVC == analysisViewController && operation == .pop {
+            // Going directly from the analysis to the camera means that
+            // the document is not an image and should be removed
+            if toVC == cameraViewController {
```

Comment:
We will need to move scattered network related calls to dedicated classes (extensions) in the future. In the Android version we have logic scattered around, too. If we are not careful it might byte us in the future.

Can you create a ticket for this? For the Android version we created a [similar one](https://github.com/gini/gini-vision-lib-android/issues/190).

Reply:
I'd like to second that comment. I was really hoping to avoid such code but I keep seeing it.

Reply:
But there is not networking here. Which makes me think that the name for that elements is still wrong. I created [a ticket](https://github.com/gini/gini-vision-lib-ios/issues/219) in order to find a better name.